### PR TITLE
Add more browser test

### PR DESCRIPTION
**Pull request by Gusted**
- Ability to restore a style when it was removed.
- Ability to watch for a source style and link positions.
- Ability to remove override when style was removed.
- Ability to react on dynamic style creation.
- Ability to react on inserting something into a style dynamically (sheet.insertRule(...)).

### CODE REVIEW

File: tests/browser/dynamic/style-override.tests.ts [hunk 1]
```diff
@@ -48,4 +48,160 @@ describe('Style override', () => {
...
         await expect(page.evaluate(() => getComputedStyle(document.querySelector('h1 strong')).color)).resolves.toBe('rgb(255, 26, 26)');
     });
+
+    it('should restore override', async () => {
+        await loadTestPage({
+            '/': multiline(
+                '<!DOCTYPE html>',
+                '<html>',
+                '<head>',
+                '   <script>',
```
Comment (alexanderby) [inline tests/browser/dynamic/style-override.tests.ts:58]: I would suggest to use `page.evaluate()` where possible, e.g.
```javascript
await page.evaluate(() => {
    const styleElement = document.createElement("style");
    styleElement.classList.add("testcase-style");
    document.head.append(styleElement);
    styleElement.sheet.insertRule("h1 { color: gray } ")
    styleElement.sheet.insertRule("strong { color: red } ")
})`
```

Please note that the function passed as an argument is converted to a string, and is executed in the page using `eval()`, follow the docs if you need to pass something into this function.

Don't forget about `code-style`

**Review by alexanderby — CHANGES_REQUESTED** [review]

File: tests/browser/dynamic/style-override.tests.ts (hunk 1 above)
Comment (Gusted) [reply tests/browser/dynamic/style-override.tests.ts:58]: Oh woops. That was something for testing purposes...

**Comment by Gusted** [comment]
Also should `npm run test` includes browser tests?

**Comment by Gusted** [comment]
Oh yeah, we have Github Actions...

**Review by alexanderby — APPROVED** [review]
I think it's okay.

**Pull request merged**